common/configdir.c

```diff
@@ -22,15 +22,22 @@ static char *default_configdir(const tal_t *ctx)
 	return path;
 }
 
+static char *default_rpc_file(const tal_t *ctx)
+{
+	char *path;
+	path = path_basename(ctx, "lightning-rpc");
+	return path;
+}
+
 void configdir_register_opts(const tal_t *ctx,
-			     char **configdir, char **rpc_filename)
+				 char **configdir, char **rpc_filename)
 {
 	*configdir = default_configdir(ctx);
-	*rpc_filename = "lightning-rpc";
+	*rpc_filename = default_rpc_file(ctx);
```

Comment:
`*rpc_filename = tal_strdup(ctx, "lightning-rpc");`

That's the one-line fix :)